internal/api/core/routing_test.go

```diff
@@ -24,47 +53,84 @@ func TestAPIPanic(test *testing.T) {
 		return nil, nil
 	}
 
-	routes = append(routes, NewAPIRoute(endpoint, handler))
+	routes = append(routes, MustNewAPIRoute(endpoint, handler))
 
 	response := SendTestAPIRequest(test, endpoint, nil)
 	if response.Locator != "-001" {
-		test.Fatalf("Response does not have panic locator of '-501', actual locator: '%s'.", response.Locator)
+		test.Fatalf("Response does not have panic locator of '-001', actual locator: '%s'.", response.Locator)
 	}
 }
 
 // Test handlers that do not have the correct signature.
 // Specifically, we are focusing on testing validateAPIHandler().
 func TestMalformedHandlers(test *testing.T) {
```

Comment:
This test seems too complex in general.
Instead, why not just change log.Fatal():
https://github.com/edulinq/autograder-server/blob/main/internal/log/public.go#L35
It uses os.Exit(), but now-a-days everyone should be using util.Exit():
https://github.com/edulinq/autograder-server/blob/main/internal/util/exit.go#L14
Then you can just see if an exit happened.

The string/output checking seems a bit too complex as well.
We really just need to see if an exit happened.
Checking the string would be nice, but not if it makes the test complex.